go/streamlog/streamlog_test.go

```diff
@@ -266,40 +266,29 @@ func TestFile(t *testing.T) {
 }
 
 func TestShouldSampleQuery(t *testing.T) {
-	queryLogSampleRate = -1
-	assert.False(t, shouldSampleQuery())
+	qlConfig := QueryLogConfig{sampleRate: -1}
+	assert.False(t, qlConfig.shouldSampleQuery())
 
-	queryLogSampleRate = 0
-	assert.False(t, shouldSampleQuery())
+	qlConfig.sampleRate = 0
+	assert.False(t, qlConfig.shouldSampleQuery())
 
-	// for test coverage, can't test a random result
-	queryLogSampleRate = 0.5
-	shouldSampleQuery()
+	qlConfig.sampleRate = 1.0
+	assert.True(t, qlConfig.shouldSampleQuery())
 
-	queryLogSampleRate = 1.0
-	assert.True(t, shouldSampleQuery())
-
-	queryLogSampleRate = 100.0
-	assert.True(t, shouldSampleQuery())
+	qlConfig.sampleRate = 100.0
+	assert.True(t, qlConfig.shouldSampleQuery())
 }
 
 func TestShouldEmitLog(t *testing.T) {
-	origQueryLogFilterTag := queryLogFilterTag
-	origQueryLogRowThreshold := queryLogRowThreshold
-	origQueryLogSampleRate := queryLogSampleRate
-	defer func() {
-		SetQueryLogFilterTag(origQueryLogFilterTag)
-		SetQueryLogRowThreshold(origQueryLogRowThreshold)
-		SetQueryLogSampleRate(origQueryLogSampleRate)
-	}()
-
 	tests := []struct {
 		sql              string
 		qLogFilterTag    string
```

Comment:
An idea: it might be cleaner to use an `expected QueryLogConfig` sub-struct

Reply:
The test is validating if logs should be emitted or not. These are all input values to QueryLogConfig.